hypershift-operator/controllers/nodepool/kubevirt/imagecacher.go

```diff
@@ -229,21 +257,45 @@ func (qi *cachedQCOWImage) buildDVForCache(nodePool *hyperv1.NodePool, infraId s
 	return dv
 }
 
-func getCacheDVSelector(infraId string) client.MatchingLabels {
+func getCacheDVSelector(uid string) client.MatchingLabels {
 	return map[string]string{
 		bootImageDVLabelRoleName: bootImageDVLabelRoleValue,
-		bootImageDVLabelInfraId:  infraId,
+		bootImageDVLabelUID:      uid,
 	}
 }
 
-func getCacheDVs(ctx context.Context, cl client.Client, infraId string, namespace string) ([]v1beta1.DataVolume, error) {
+func getCacheDVs(ctx context.Context, cl client.Client, uid string, namespace string) ([]v1beta1.DataVolume, error) {
 	dvs := &v1beta1.DataVolumeList{}
 
-	err := cl.List(ctx, dvs, client.InNamespace(namespace), getCacheDVSelector(infraId))
+	err := cl.List(ctx, dvs, client.InNamespace(namespace), getCacheDVSelector(uid))
 
 	if err != nil {
 		return nil, fmt.Errorf("failed to read DataVolumes; %w", err)
 	}
 
 	return dvs.Items, nil
 }
+
+// DeleteCache deletes the cache DV
+//
+// This function is not part of the interface, because it called from the nodePool reconciler Delete() method, that is
+// called before getting the cacheImage.
+func DeleteCache(ctx context.Context, cl client.Client, name, namespace string) error {
+	dv := v1beta1.DataVolume{}
+	err := cl.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, &dv)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil // already deleted
+		}
+
+		return fmt.Errorf("failed to get DataVolume %s/%s: %w", namespace, name, err)
+	}
+
+	err = cl.Delete(ctx, &dv)
+	if err != nil {
+		return fmt.Errorf("failed to delete DataVolume %s/%s: %w", namespace, name, err)
+	}
```

Comment:
Usually we check to see if `dv.DeletionTimestamp != nil` before calling delete on an obj already marked for deletion.